paths/paths_unix.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build !windows
+
+package paths
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"golang.org/x/sys/unix"
+)
+
+func init() {
+	stateFileFunc = stateFileUnix
+}
+
+func stateFileUnix() string {
+	// TODO: use other default paths on other GOOSes probably. This works for Linux.
+	const varLib = "/var/lib/tailscale/tailscaled.state"
+	try := varLib
+	for i := 0; i < 3; i++ { // check writability of the file, /var/lib/tailscale, and /var/lib
```

Comment:
Why does /var/lib need to be writable? That will definitely be false when I lock down the systemd configs to use a private state dir.

Reply:
It doesn't need to be.

Look at line 27/28. It works its way up and returns once it finds out it has permissions.

So if you're running as root (normal user running sudo without using our packages or systemd lock down), it'll learn that up at /var/lib being writable, which means it can then create /var/lib/tailscale automatically later.

Reply:
Still very not a fan of daemons scribbling their own directories into /var/lib when they don't like how things look, but okay, I guess.

Reply:
I don't understand.

By scribbling do you mean creating a directory named after the name of the program, like nearly every program in existence does?

Reply:
Yes, I do mean that. Just because many programs do things I dislike doesn't mean I have to be happy about us doing it as well :)

But it makes you happy significantly more than it makes me unhappy, so it's a net win.